### DISCUSSION TITLE
certificate documentation improval notes

### DISCUSSION THREAD

**Discussion by ennerf** [hydraulic-software/conveyor#22] [General]
First of all, kudos for simplifying the certificate process with provided certificate requests and the ability to work directly with the `.cer` files rather than exporting a `p.12`. This is the first time that mac signing didn't feel extremely painful 👍 

A few things to improve the docs (https://conveyor.hydraulic.dev/1.1/keys-and-certificates/):

* The "Apple ID account webpage" should have a link to https://appleid.apple.com/account/manage/section/security to avoid googling
* The DigiCert link should go directly to the relevant page https://www.digicert.com/dc/code-signing/microsoft-authenticode.htm
* Maybe add a sentence explaining the difference between normal and EV certificates. Are EV the same but without the "Do you want to trust code from company X?" prompt? Are there huge differences in the approval process that would be good to know beforehand?

It'd be convenient to allow base64 versions `.cer` files so CI secrets / env variables can be mapped directly without adding actions to create temporary files. Ideally with a documentation snippet for creating a copy & paste-able CI secret (e.g. https://www.igorkromin.net/index.php/2017/04/26/base64-encode-or-decode-on-the-command-line-without-installing-extra-tools-on-linux-windows-or-macos/)

**Comment by mikehearn** [comment]
@ogpublic Another one for you (internal link: [CO-210](https://youtrack.hq.hydraulic.software/issue/CO-210)).

The difference between EV certs and normal is actually already explained in the signing FAQ section here:

https://conveyor.hydraulic.dev/1.0/faq/signing-and-certificates/#5-whats-the-difference-between-a-normal-and-ev-certificate

We need to make the info more visible, clearly. Could add a link to the FAQ entry, or move the FAQ answer to be a section in the keys-and-certs page and leave a link in the FAQ. Ovi - up to you how to best resolve this. 

Base64 CER files should work already. However, certificates are public so there's no need to hide them. The only thing that's sensitive is the keys. If you're letting Conveyor derive keys from the root entropy for you, the `app.signing-key` value is text already and can be moved into either a secret file or a secret environment variable. There's a section in the docs on how to protect secrets when using CI here:

https://conveyor.hydraulic.dev/1.0/keys-and-certificates/#continuous-integration

Let us know if that description of how to set up CI is useful. Maybe you didn't spot it because it's at the bottom of the page? We should probably link to it from the end of the tutorial.

**Comment by ennerf** [comment]
> Could add a link to the FAQ entry

A link to the FAQ next to the DigiCert link would be great

> Let us know if that description of how to set up CI is useful.

I apparently stopped reading at "Hardware Security Modules" and thought it's not relevant for me anymore.  The CI section is actually quite good, but I think it should come right after the Example and emphasize that the `.cer` files are public and can be left with source code. Exporting keys and HSM seem like advanced use cases that are probably irrelevant for most users.

**Comment by mikehearn** [comment]
We implemented these improvements in Conveyor 2!
